[nativert] Port string join and split to c10/util

yiming0416 opened the pull request:
Summary: 
Torch Native Runtime RFC: https://github.com/pytorch/rfcs/pull/72
Port string utils functions join and split to c10/util

Test Plan:
Added tests in `string_util_test.cpp`
buck2 run mode/opt caffe2/c10/test:util_base_tests

Differential Revision: D74202473

cyyever commented on c10/util/StringUtil.h:
```diff
@@ -238,6 +239,13 @@ C10_API std::optional<double> tryToNumber<double>(const char* symbol);
+
+C10_API std::string join(
+    std::string_view delimiter,
+    const std::vector<std::string>& keys);
```
Is it better to use `std::vector<std::string_view>`?

yiming0416 replied:
@cyyever  well maybe... but for usability, we have existing internal usage of this function with `std::vector<std::string>` as the argument type, it might be a bit clunky to convert the input types. So I'd prefer to keep the current signature if this is not a blocking comment.

cyyever replied:
Fine

cyyever commented:
@yiming0416 If there are more string functions coming, we should consider ranges.

yiming0416 committed:
[nativert] Port string join and split to c10/util (#152873)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/152873

Port string utils functions join and split to c10/util

Test Plan:
Added tests in `string_util_test.cpp`
buck2 run mode/opt caffe2/c10/test:util_base_tests

Differential Revision: D74202473

yiming0416 commented:
Removed the newly added `join()` as it can be replaced by `fmt::join()`

huydhn commented:
@pytorchbot merge
